pkg/process/util/orchestrator/cache.go

```diff
@@ -0,0 +1,42 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// +build orchestrator
+
+package orchestrator
+
+import (
+	"time"
+
+	cache "github.com/patrickmn/go-cache"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	defaultExpire = 5 * time.Minute
+	defaultPurge  = 30 * time.Second
+)
+
+// KubernetesResourceCache provides an in-memory key:value store similar to memcached for kubernetes resources.
+var KubernetesResourceCache = cache.New(defaultExpire, defaultPurge)
```

Comment:
Why not re-using `cache.Cache` instance?

Reply:
I reused our cache initially.
@mfpierre  @ahmed-mez and I were talking about this in slack and decided that using our own cache would be the way to go. One reason was because of the large uid keyset and we wanted to have control over the cache.